database/data.js

```diff
@@ -0,0 +1,216 @@
+const LoremIpsum = require("lorem-ipsum").LoremIpsum;
+
+const lorem = new LoremIpsum({
+  sentencesPerParagraph: {
+    max: 8,
+    min: 4
+  },
+  wordsPerSentence: {
+    max: 16,
+    min: 4
+  }
+});
+
+//FUNCTIONS
+//need function to create a sentence for the name of item + color at end
+var createItemName = () => {
+  var itemName = null;
+  //an item name should contain between 5 - 10 words + a color
+  //randomize a number between 5 - 10 words
+  var numberOfWords = Math.floor(Math.random() * (10 - 5)) + 5;
+  //need to call lorem.generateSentences(1) + color
+  // itemName = lorem.generateWords(numberOfWords) + `, ${color}`;
+  itemName = lorem.generateWords(numberOfWords);
+  //return item Name
+  return itemName;
+};
+
+//need function to use lorem ipsum to generate 5-7 paragraphs for description
+var createItemDescription = () => {
+  //create an array
+  var itemDescription = [];
+  //create random number between 4-7
+  var max = Math.floor(Math.random() * (7 - 4)) + 4;
+  //set max number = to random number above
+  //loop from 0 to max number
+  for (var i = 0; i <= max; i++) {
+    //for each loop call lorem ipsum to create one paragraph
+    var paragraph = lorem.generateParagraphs(1);
+    //push paragraph into array
+    itemDescription.push(paragraph);
+  }
+  //return array
+  return itemDescription;
+};
+
+//need function to select random brand from a list of speakers
+var getBrand = (num) => {
+  //input: a number between 1000- 1100
+  //output: a string
+  //brands "Nile", "Mississippi", "Thames", "Ganges", "Danube", "Yangtze"
+  if (num < 1015) {
+    return 'Yangtze';
+  }
+  if (num < 1030) {
+    return 'Mississippi';
+  }
+  if (num < 1045) {
+    return 'Thames';
+  }
+  if (num < 1060) {
+    return 'Ganges';
+  }
+  if (num < 1075) {
+    return 'Danube';
+  }
+  return 'Nile';
+};
+
+//need function to select either true or false randomly
+var generateBooleanValue = () => {
+  var num = Math.round(Math.random());
+  if (num === 0) {
+    return false;
+  } else {
+    return true;
+  }
+};
+
+//Create Configuration data
+var createConfiguration = () => {
+  //return an array of 4 values
+  var configArray = [];
+  //each value can be a few words to no words
+  //randomize numbers between 1- 5 to get a number of configurations for the item
+  var numOfConfigs = Math.floor(Math.random() * (5 - 1)) + 1;
+
+  //loop from 0 - numOfConfigs
+  for (var i = 0; i < numOfConfigs; i++) {
+    //randomize a number between 1-4  to get a number of words for each config
+    var numOfWords = Math.floor(Math.random() * (4 - 1)) + 1;
+    //call generateWords with random number above
+    var config = lorem.generateWords(numOfWords);
+    //push generated words into array
+    configArray.push(config);
+  }
+  //return array
+  return configArray;
+};
+
+//need function to keep track of similar items based on color
+var getAvailableColors = (id, color) => {
+  //input is a color and id
+  var arrayOfProductIds = [];
```

Comment:
I like this idea. It was a bit confusing to understand at first. Maybe consider being a bit more specific with the variable name here. Maybe `arrayOfSimilarProductIds` would be a little easier to understand.